cpp/src/io/orc/writer_impl.cu

```diff
@@ -1073,23 +1073,49 @@ void set_stat_desc_leaf_cols(device_span<orc_column_device_view const> columns,
                    [=] __device__(auto idx) { stat_desc[idx].leaf_column = &columns[idx]; });
 }
 
-writer::impl::encoded_statistics writer::impl::gather_statistic_blobs(
-  statistics_freq stats_freq,
+hostdevice_vector<uint8_t> allocate_and_encode_blobs(
+  hostdevice_vector<statistics_merge_group>& stats_merge_groups,
+  rmm::device_uvector<statistics_chunk>& stat_chunks,
+  int num_stat_blobs,
+  rmm::cuda_stream_view stream)
+{
+  // figure out the buffer size needed for protobuf format
+  gpu::orc_init_statistics_buffersize(
+    stats_merge_groups.device_ptr(), stat_chunks.data(), num_stat_blobs, stream);
+  stats_merge_groups.device_to_host(stream, true);
```

Comment:
It looks like only `stats_merge_gropus[num_stat_blobs - 1]` needs to be read from this device-to-host copy. Is it possible to copy only one value (or even just the `start_chunk` and `num_chunks`) instead of the entire vector?

Reply:
There is a good deal of device to host and back. I kept having to reel myself in here as I didn't want to drastically alter this. There isn't a way to get a `rmm::device_buffer` element to the host and `rmm::device_uvector` requires removing the default constructor. So for now I'm going to lift the element function from `rmm::device_uvector`, but I don't like that either. I guess it is a step in the right direction...